File: android/beacon/9-Notification-Strategy/Starter/app/src/main/java/android/connecthings/com/tuto/notification/BeaconNotificationStrategyFilter.java
package android.connecthings.com.tuto.notification;

import android.connecthings.adtag.adtagEnum.FEED_STATUS;
import android.connecthings.adtag.model.sdk.BeaconContent;
import android.connecthings.util.adtag.beacon.strategy.BeaconNotificationStrategy;

/**
 */
public class BeaconNotificationStrategyFilter implements BeaconNotificationStrategy {

    private long minNextTimeNotification;
    private int minTimeBetweenNotification;

    public BeaconNotificationStrategyFilter(int minTimeBetweenNotification) {
        this.minTimeBetweenNotification = minTimeBetweenNotification;
    }

    @Override
    public boolean deleteCurrentNotification(BeaconContent beaconContent, FEED_STATUS feed_status) {
        return minNextTimeNotification < System.currentTimeMillis();
    }

    @Override
    public boolean createNewNotification(BeaconContent beaconContent, FEED_STATUS feed_status) {
        return false;
    }

    @Override
    public void onNotificationCreated(boolean b, BeaconContent beaconContent) {

    }

    @Override
    public void onNotificationDeleted(boolean b, BeaconContent beaconContent, FEED_STATUS feed_status) {
        if(b){
            minNextTimeNotification = System.currentTimeMillis() + minTimeBetweenNotification;
        }
    }

    @Override
    public void onStartMonitoringRegion(BeaconContent beaconContent, FEED_STATUS feed_status) {

    }

    @Override
    public void onBackground() {

    }

    @Override
    public void onForeground() {

    }
}

File: ios/ObjectiveC/Beacon/6-Notification-Strategy/Complete/NotificationStrategyComplete/BeaconNotificationStrategyFilter.m
//
//  BeaconNotificationStrategyFilter.m
//  Notification
//
//

#import "BeaconNotificationStrategyFilter.h"

@implementation BeaconNotificationStrategyFilter

-(NSString *) getName {
    return @"beaconNotificationStrategyFilter";
}

-(void)updateParameters:(NSDictionary *) params {
}

-(id) initWithMinTimeBetweenNotification:(long)minTime{
    self = [super init];
    if(self){
        minTimeBetweenNotification = ((double) minTime / 1000);
        minNextTimeNotification = 0 ;
    }
    return self;
}

-(BOOL) createNewNotification:(ATBeaconContent *)newBeaconContent feedStatus:(ATRangeFeedStatus)feedStatus{
    return minNextTimeNotification < CACurrentMediaTime();
}

-(void) onNotificationIsCreated:(ATBeaconContent *)beconContent notificationStatus:(BOOL)notificationStatus{
    
}

-(BOOL) deleteCurrentNotification:(ATBeaconContent *)newBeaconContent feedStatus:(ATRangeFeedStatus)feedStatus{
    return true;
}

-(void) onNotificationIsDeleted:(ATBeaconContent *)beconContent notificationStatus:(BOOL)notificationStatus{
    if(notificationStatus){
        minNextTimeNotification = CACurrentMediaTime() + minTimeBetweenNotification;
    }
}

-(void) onBackground{}

-(void) onForeground{}

-(void) onStartMonitoringRegion:(ATBeaconContent *)beaconContent feedStatus:(ATRangeFeedStatus)_feedStatus{}

-(void) load:(NSUserDefaults *)dataStore{
    minNextTimeNotification = [dataStore doubleForKey:MIN_NEXT_TIME_NOTIFICATION];
}

-(void) save:(NSUserDefaults *)dataStore{
    [dataStore setDouble:minNextTimeNotification forKey:MIN_NEXT_TIME_NOTIFICATION];
}

@end
